### DISCUSSION TITLE
Custom titles for clipped tiddler

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#46] [Q&A] [answered]
Is it possible to get custom titles for the tiddlers created by clipping. By default, clipped tiddler title will be the name of the webpage. 
Is it possible to change the rule tiddler in such a way that a dialogue box pop ups to enter the tiddler title when the clipping is being done (similar to the pin and prose concept). I had read the documentation regarding Tiddlers titles, but couldn't find a solution for this.

<img width="658" alt="image" src="https://user-images.githubusercontent.com/67494083/149212281-59bc31d6-301c-4e0d-81b4-27b79c653ec4.png">

**Comment by buggyj** [comment, marked as answer]
You can use a 'userstring' for this. eg you could modify the 'snip' entry in the TiddlyClipConfig like so

![image](https://user-images.githubusercontent.com/2855795/149291607-892ceb2f-71b9-4045-9052-23d48d434e38.png)

this instructs the TiddlyClip addon to show a dialogue box, when the 'snip' rule is clicked.

The data input from the dialogue box is sent the the tiddlywiki in the variable '@userstring'. You can then use this in the snip rule for the tiddler name like so:

![image](https://user-images.githubusercontent.com/2855795/149293301-44e24931-e386-4853-8cc5-c1a7ed601d34.png)

**Reply by arunnbabu81** [reply]
Thanks for the detailed answer. 

When using userstring for tiddler titles, if I want to clip multiple non contigous text from the same webpage into the same tiddler(whose title was assigned using userstring), how can I do this ? 

1.  I used the same userstring every time I clipped from the same webpage and the clipped texts were saved in the same tiddler. 
2. I also tried to select multiple non contigous text , all at the same time  by holding "Control" key while selecting. Here also the clipped texts were saved in the same tiddler, but without any formatting
 
Is there any other way to do this?

**Reply by buggyj** [reply]
Firstly, webpages are complex, and a clipping a part of a webpage does not always retain the same formatting as in the original.

Tiddlyclip is designed to be programmed so there are in fact many ways to organize multiple clips from the same page. I use a number of different ways depending on what I am trying to do.

One simple way is to clip a larger area of a webpage and use the ckeditor plugin to modify the tiddler that contains the clip.

It is also possible to edit the webpage in the browser before clipping it - they are addons to do this (https://addons.mozilla.org/en-US/firefox/addon/page-edit-design-mode/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search)
